app/forms/profile_form.rb

```diff
@@ -0,0 +1,67 @@
+class ProfileForm
+  include ActiveModel::Model
+  include Virtus.model
+
+  AFFILIATION_VALUES = [ '大学生', '大学院生', '高専生', '専門学生', '社会人', 'その他' ]
+
+  validates :affiliation, inclusion: { in: AFFILIATION_VALUES }
+  validates :school,     length: { maximum: 50 }
+  validates :faculty,    length: { maximum: 50 }
+  validates :department, length: { maximum: 50 }
+  validates :laboratory, length: { maximum: 50 }
+  validates :content,    length: { maximum: 200 }
+  validate :at_least_one_parameter
+
+  attribute :affiliation, String
+  attribute :school,      String
+  attribute :faculty,     String
+  attribute :department,  String
+  attribute :laboratory,  String
+  attribute :content,     String
+  attribute :user_id,     Integer
+
+  attr_accessor :profile
+
+  def initialize(profile = Profile.new)
+    @profile = profile
+    # プロフィールが保存済みならProfileFormのattributesに代入
+    self.attributes = @profile.attributes if profile.persisted?
+  end
+
+  def assign_attributes(params = {})
+    @params = params
+    # ユーザーがすでにprofileを持っている場合、profileのattributesを更新
+    profile.assign_attributes(profile_form_params) if profile.persisted?
+    # 更新情報をバリデーションするため、ProfileFormのattributesも変更
+    super(profile_form_params)
```

Comment:
`@params`に格納する必要がなくなります。
```suggestion
    # ユーザーがすでにprofileを持っている場合、profileのattributesを更新
    profile.assign_attributes(params) if profile.persisted?
    # 更新情報をバリデーションするため、ProfileFormのattributesも変更
    super(params)
```